HPCC-30280 Add missing index read stats to the Thor mapping

ghalliday opened the pull request:
<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

ghalliday committed:
HPCC-30280 Add missing index read stats to the Thor mapping

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday commented:
This fixes several problems:
- blob load time and other stats were not in the mapping, so they were never reported
- httpcall would fail in debug because the context did not have the StTimeSoapcall stat
- ContextLogger now has the mapping passed in to it.
- soapcall now reports the time in the soapcall (previously it was probably thrown away)

After this change I think there is more scope for rationalisation.  In particular, the CStatsCtxLoggerDeltaUpdater probably does not need the mapping passed to it, and it be simpler if its functionality was actually merged into the context logger.

There may be some calls to resetStart missing.

ghalliday commented on thorlcr/activities/soapcall/thsoapcallslave.cpp:
```diff
@@ -296,15 +308,17 @@ class SoapDatasetActionSlaveActivity : public ProcessSlaveActivity, implements I
         wscHelper->start();
         wscHelper->waitUntilDone();
         IException *e = wscHelper->getError();
+        wscHelper.clear();
```
I suspect this is unnecessary - the problem I was trying to fix was elsewhere, but probably harmless.

captain-hoity-toity commented on thorlcr/activities/keyedjoin/thkeyedjoinslave-legacy.cpp:
```diff
@@ -1253,7 +1253,7 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
     public:
```
Have you run the integration tests?

ghalliday commented:
@GordonSmith this ran out of disk space.  Is that a new problem?

jakesmith commented on system/jlib/jstats.cpp:
```diff
@@ -2439,7 +2440,8 @@ void CRuntimeStatisticCollection::set(const CRuntimeStatisticCollection & other,
         StatisticKind kind = other.getKind(i);
         unsigned __int64 value = other.getStatisticValue(kind);
-        setStatistic(kind, value, node);
+        if (value)
```
not sure about this check, why it's needed, If all the stat kind's are now correctly in the target.
Wondered if it might cause problems by not setting if 0 in some situations.

jakesmith reviewed (commented):
@ghalliday - looks good, but 1 comment/question.

GordonSmith commented:
> @GordonSmith this ran out of disk space. Is that a new problem?

Just a temporary issue, until vcpkg auto builds catch up with GH builder changes.

ghalliday commented:
@jakesmith I removed the line and squashed.  Currently rerunning to check if any asserts are being generated - hence the set of new jiras for other asserts that are being hit.

jakesmith reviewed (approved):
@ghalliday - looks good.
